wasm/read.go

```diff
@@ -25,3 +26,21 @@ func readString(r io.Reader, n int) (string, error) {
 	}
 	return string(bytes), nil
 }
+
+func readU32(r io.Reader) (uint32, error) {
+	var buf [4]byte
+	_, err := io.ReadFull(r, buf[:])
+	if err != nil {
+		return 0, err
+	}
+	return binary.LittleEndian.Uint32(buf[:]), nil
+}
+
+func readU64(r io.Reader) (uint64, error) {
+	var buf [4]byte
```

Comment:
Followup question, how come tests didn't catch this? I'd expect things to fail badly as a result of a low level primitive like this being incorrect. Is this code not used/tested significantly yet? (That's completely fine if so, I just want to learn.)